App.tsx

```diff
@@ -474,6 +475,22 @@ const BaseApp: React.FunctionComponent<{
     );
   }
 
+  let initialUrl: string | null = null;
+  // get the universal link the app was open with, if one exists
+  Linking.getInitialURL()
+    .then(url => {
+      if (url) {
+        initialUrl = url;
+      }
+    })
+    .catch(err => {
+      logger.error('An error occurred while getting InitialUrl.', err);
+    });
+
+  Linking.addEventListener('url', event => {
```

Comment:
I verified that this listened update the initial Url and things work as expected 😄 (did it in a hacky way, but you know, it was easy to test lol see screenshot) this also made it easy to see what the initialUrl had in it. 
![IMG_1031](https://github.com/user-attachments/assets/ec59c7f9-fde2-4cc0-9b65-19d06a1e63d0)
![IMG_1032](https://github.com/user-attachments/assets/e9947c3d-ed0e-449e-92e2-b052841b9527)

Reply:
This is an interesting approach. It makes me queasy simply because I don't think we have any guarantee in what order the event loop handles your listener versus the `getStateFromPath()` call. The benefit of continuing to use React Navigation linking as opposed to having to hand-roll it might be large enough that the benefits here outweigh the cons, but fundamentally I do worry that out-of-order executions here would totally break links. You know how much we struggled with this sort of thing on the `onLayout` hacks to try to make the map react to the size of the header. 90% of the time, it works every time. The other 10%, it just doesn't and we have nothing we can do to fix it. When the outcome of that is that the map overlaps weirdly, whatever. When the outcome is that opening a link doesn't work, not sure if that's ok.

Reply:
I think your concern is valid - although from testing/reading about this it seems like this is the suggested and trusted way to get the url. 🤷 Fair to say that we wont know when the 10% fail happens until we get a wider use of this.

I am not sure I get how we can use react navigation here: how do you see navigation linking providing the info that can get us center id? Aside from either having the center id in the url or querying NAC using the observation Id I dont see another way. Or after loading the obs page and then updating the logo in the header or something...or just not having the logo in the shared obs altogether.

Reply:
chatted about this offline - this works for just obs sharing, but gets way more complex when we will share forecasts, weather, etc. For now, we will keep as is and watch for errors --- but in future may take a different approach (i.e. not having linking be generated for us)